### DISCUSSION TITLE
[2023-03-23] Caravel-SoC Testbench and Firmware (Updated with Code Annotation)

### DISCUSSION THREAD

**Discussion by kevinjantw** [bol-edu/caravel-soc#19] [General]
[caravel-soc_testbench&firmwave.pdf](https://github.com/bol-edu/caravel-soc/files/11052078/caravel-soc_testbench.firmwave.pdf)

**Comment by wei851246721** [comment]
<img width="895" alt="Snipaste_2023-09-19_22-19-36" src="https://github.com/bol-edu/caravel-soc/assets/59242319/180e8cc3-a60b-4a96-95c1-8c36c87d2f81">
hi,请问我打开Jupyter Notebook的路径总是在/，该如何修改打开Jupyter Notebook的路径呢？

我想利用lab1的.bit 与.hwh执行Jupyter 的操作，但总是失败的，似乎是路径导致的，Jupyter 的路径总是/，无法调整

**Comment by boledulab** [comment]
可以參考這個例子
https://github.com/bol-edu/caravel-soc_fpga#validation-on-pynq-fpga-board

Jupyter Notebook路径/ 不需要更改 (對應到python code裡/home/xilinx/jupyter_notebooks/)
直接在這路径上傳xx.bit与xx.hwh
在python code裡改成 /home/xilinx/jupyter_notebooks/xx.bit

**Reply by wei851246721** [reply]
Thanks！ it's  ok.
